addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -77,31 +77,38 @@ function insert(editor, data, isText = true) {
     }
     let nodeToInsert;
     const insertedNodes = [...fakeEl.childNodes];
+    let nodeToInsertId = 0;
     while ((nodeToInsert = fakeEl.childNodes[0])) {
         if (isBlock(nodeToInsert) && !allowsParagraphRelatedElements(startNode)) {
-            // Split blocks at the edges if inserting new blocks (preventing
-            // <p><p>text</p></p> scenarios).
-            while (
-                startNode.parentElement !== editor.editable &&
-                !allowsParagraphRelatedElements(startNode.parentElement)
-            ) {
-                if (isUnbreakable(startNode.parentElement)) {
-                    makeContentsInline(fakeEl);
-                    nodeToInsert = fakeEl.childNodes[0];
-                    break;
-                }
-                let offset = childNodeIndex(startNode);
-                if (!insertBefore) {
-                    offset += 1;
-                }
-                if (offset) {
-                    const [left, right] = splitElement(startNode.parentElement, offset);
-                    startNode = insertBefore ? right : left;
-                } else {
-                    startNode = startNode.parentElement;
+            if (nodeToInsertId === 0) {
+                makeContentsInline(nodeToInsert);
+                nodeToInsert = nodeToInsert.childNodes[0];
+            } else {
```

Comment:
So all that changed here is this base case right? What does it cover? Could you add a comment?